File: arrays_strings/arrays_strings.py
import math
from os import stat


def reverseString(s):
    i = 0
    j = len(s) - 1
    while i < j:
        temp = s[i]
        s[i] = s[j]
        s[j] = temp
        i+=1
        j-=1
    return s

# print(reverseString(["h", "e", "l", "l", "o"]))

def maxArea(list):
    maxarea = 0
    i = 0
    j = len(list) - 1
    
    while i < j:
        width = j - i
        area = min(list[i], list[j]) * width
        maxarea = max(maxarea, area)
        if list[i] <= list[j]:
            i += 1
        else:
            j -= 1
            
    return maxarea
print( maxArea([1,8,6,2,5,4,8,3,7]) )

File: arrays_strings/arrays_strings.js
var reverseString = function(s) {
    var left = 0, right = s.length - 1;
    while (left < right) {
        var temp = s[left];
        s[left++] = s[right];
        s[right--] = temp;
    }
    return s;
};
console.log( reverseString(['h', 'e', 'l', 'l', 'o']) )

const maxArea = arr =>{
    // 1. Initialize max
    var max=0;
    // 2. For left side loop
        for (var i=0; i<arr.length; i++){
        // 3. For riht side loop starting later
            for (var j=i+1; j<arr.length; j++){
                // 4. Get the min of i and j multiply tat by j-i
                var area = Math.min(arr[i], arr[j]) * (j-i)
                // 5. Get max of current area and current max
                max = Math.max(max, area)   
            }
        }
    return max;
}
console.log(maxArea([1,8,6,2,5,4,8,3,7]))